### PR TITLE
Support new elixir-graphql query execution style.

### PR DESCRIPTION
**Pull request by plasticine**
Adds support for joshprice/graphql-elixir#25

~~Obviously joshprice/graphql-elixir#25 is unmerged as yet, but I wanted to have a play with it — hope this is useful :)~~ Merged! :dancer:

### CODE REVIEW

File: lib/graphql/plug/endpoint.ex [hunk 1]
```diff
@@ -41,8 +39,22 @@ defmodule GraphQL.Plug.Endpoint do
     |> halt
   end
 
-  defp execute(conn, schema, query) do
-    case GraphQL.execute(schema, query) do
+  defp parse(query) do
```
Comment (joshprice) [inline lib/graphql/plug/endpoint.ex]: Shouldn't need this any more as `GraphQL` module only exposes `execute` function

File: lib/graphql/plug/endpoint.ex [hunk 2]
```diff
@@ -41,8 +39,22 @@ defmodule GraphQL.Plug.Endpoint do
...
+
+  defp execute(conn, _, {:error, errors} = _) do
+    case Poison.encode(errors) do
+      {:ok, json}      -> send_resp(conn, 400, json)
+      {:error, errors} -> send_resp(conn, 400, errors)
+    end
+  end
+
+  defp execute(conn, schema, document) do
+    case GraphQL.execute(schema, document) do
```
Comment (joshprice) [inline lib/graphql/plug/endpoint.ex]: This is now `GraphQL.Execution.Executor.execute`

File: mix.exs [hunk 3]
```diff
@@ -28,7 +28,7 @@ defmodule PlugGraphql.Mixfile do
      {:cowboy, "~> 1.0"},
      {:plug, "~> 0.14 or ~> 1.0"},
      {:poison, "~> 1.5"},
-     {:graphql, "~> 0.0.4"}]
+     {:graphql, github: "joshprice/graphql-elixir", branch: "execute-queries"}]
```
Comment (joshprice) [inline mix.exs]: Once I've merged the latest change upstream I'll publish a new version

File: lib/graphql/plug/endpoint.ex [hunk 4]
```diff
@@ -8,8 +8,7 @@ defmodule GraphQL.Plug.Endpoint do
     opts
   end
 
-  def call(%Conn{method: req_method, params: %{"query" => query}} = conn, opts)
-  when req_method in ["GET", "POST"] do
+  def call(%Conn{method: req_method, params: %{"query" => query}} = conn, opts) when req_method in ["GET", "POST"] do
```
Comment (joshprice) [inline lib/graphql/plug/endpoint.ex]: Is there some standard convention for this? I have aversions to ridiculously long lines... ;)

Comment (plasticine) [inline lib/graphql/plug/endpoint.ex]: @joshprice Fair enough, I’ve backed that change out for now.

Having done a quick search it seems the answer to the convention question is “there isn’t one”. I don’t love the long lines either, but broken up I actually completely missed those guard clauses initially.

**Comment by plasticine** [comment]
@joshprice Have cleaned this up ready for another look (though really nothing other than some test code has changed in the end!).

**Comment by joshprice** [comment]
:+1:

**Pull request merged**